input.go

```diff
@@ -18,6 +18,19 @@ import (
 	"github.com/hajimehoshi/ebiten/internal/ui"
 )
 
+// CopyChars copies "printable" runes read from the keyboard each update to buf.
+//
+// CopyChars represents the environment's locale-dependent translation of keyboard
+// input to Unicode characters.
+//
+// IsKeyPressed is based on a mapping of device codes to input device keys.
+// "Control" and modifier keys should be handled with IsKeyPressed.
+//
+// This function is concurrent-safe.
+func CopyChars(buf []rune) int {
```

Comment:
There's not too many examples of what's being done in the standard library and the approach taken to cgo callbacks in general is different in the x hierarchy, but reflect.Copy expresses a copy builtin happening under the hood returning its int.